source/features/batch-mark-files-as-viewed.tsx

```diff
@@ -59,6 +59,7 @@ function onAltClick(event: delegate.Event<MouseEvent, HTMLInputElement>): void {
 			? 'Marking visible files as unviewed'
 			: 'Marking visible files as viewed',
 		doneMessage: 'Marking files completed',
+		onDone: ToastOnDoneState.success,
```

Comment:
This should still be automatic, no change.  There should not be any onDone call back nor ondoneState

Reply:
I added States for better flexibility in the future. Info State is the default.

Reply:
There's no need

Reply:
Finally will be run even after catch, right?

Reply:
And I show Done in finally block

Reply:
Then drop the `finally` and keep its code after the catch:

```js
try
  return await task()
catch (error) {
  messageWrapper.textContent = error.message
  throw error // Throw it again so it can be caught again when awaiting `showToast`
}

toast.classList.replace('Toast--loading', 'Toast--success');
messageWrapper.textContent = doneMessage;
iconWrapper.firstChild!.replaceWith(<CheckIcon/>);
```

Reply:
Okay, dropping `finally{ }`.

Reply:
I pushed some changes to the toast file. It should probably not receive further changes other than fixing its types.